### DISCUSSION TITLE
Find all comments by a user with graphql v4 api

### DISCUSSION THREAD

**Discussion by sorenlouv** [community/community#24400] [Apps, API and Webhooks]
<p>In the v3 API it was possible to get all comments from a user via the events API:</p>
<pre><code>https://api.github.com/users/sqren/events
</code></pre>
<p>I cannot find a similar API where I can retrieve comments across repositories and issues. </p>
<p>If I at least could retrieve all comments by user user in specific repo, that would be great.</p>
<p>Thanks</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>A query like this should give you what you’re looking for:</p>
<pre><code>{
  user(login: "lee-dohm") {
    commitComments(first: 100) {
      nodes {
        commit {
          repository {
            nameWithOwner
          }
          abbreviatedOid
        }
        body
      }
    }
    issueComments(first: 100) {
      nodes {
        repository {
          nameWithOwner
        }
        issue {
          number
        }
        pullRequest {
          number
        }
        body
      }
    }
  }
}
</code></pre>
<p>There’s also a gistComments connection if you’re interested in that, but it sounds like you want comments within the context of a repository.</p>
<p>I hope that helps!</p>

**Comment by sorenlouv** [comment]
<p>This is terrific. Thanks a lot!</p>
